Added Italian translation to docs file index.md (#1576)
fastapi/fastapi: docs/it/docs/index.md

<p align="center">
  <a href="https://fastapi.tiangolo.com"><img src="https://fastapi.tiangolo.com/img/logo-margin/logo-teal.png" alt="FastAPI"></a>
</p>
<p align="center">

>     <em>Framework FastAPI, alte prestazioni, facile da imparare, veloce da programmare, pronto per il rilascio in produzione</em>

Reviewer: <em>Framework FastAPI, elevate prestazioni, facile da imparare, veloce da programmare, pronto per il rilascio in produzione</em>

>     <em>Framework FastAPI, alte prestazioni, facile da imparare, veloce da programmare, pronto per il rilascio in produzione</em>

Reviewer: <em>Framework FastAPI, prestazioni elevate, facile da imparare, veloce da programmare, pronto per il rilascio in produzione</em>

---

**Documentazione**: <a href="https://fastapi.tiangolo.com" target="_blank">https://fastapi.tiangolo.com</a>

**Codice Sorgente**: <a href="https://github.com/tiangolo/fastapi" target="_blank">https://github.com/tiangolo/fastapi</a>

---

> FastAPI è un web framework moderno e veloce (ad alte prestazioni) per progettare API con Python 3.6+, basato sulle annotazioni di tipo standard di Python.

Reviewer 2: Suggestion: "le annotazioni ~~di tipo~~ standard"

> FastAPI è un web framework moderno e veloce (ad alte prestazioni) per progettare API con Python 3.6+, basato sulle annotazioni di tipo standard di Python.

Reviewer: (con prestazioni elevate)

Le sue caratteristiche principali sono:

* **Veloce**: Prestazioni elevate, alla pari di **NodeJS** e **Go** (grazie a Starlette e Pydantic). [Uno dei framework in Python più veloci in circolazione](#performance).

> * **Veloce da programmare**: Velocizza il lavoro consentendoti di rilasciare nuove funzionalità tra il 200% e il 300% più velocemente. *

Reviewer 3: ```suggestion
* **Veloce da programmare**: Velocizza il lavoro consentendo il rilascio di nuove funzionalità tra il 200% e il 300% in maniera più rapida. *
```

* **Meno errori**: Riduce di circa il 40% gli erorri commessi dagli sviluppatori. *
* **Intuitivo**: Ottimo supporto per gli editor di testo con <abbr title="anche noto come IntelliSense">autocompletamento</abbr> ovunque. Perdi meno tempo a trovare errori.
* **Facile**: Progettato per essere facile da usare e imparare. Perdi meno tempo a consultare la documentazione.

> * **Breve**: Minimizza la duplicazione di codice. Multiple funzionalità con ciascuna dichiarazione di parametri. Meno errori.

Reviewer: * **Sintetico**

> * **Robusto**: Scrivi codice pronto per essere rilasciato in produzione con una documentazione dell'API interattiva generata automaticamente.

Reviewer 3: ```suggestion
* **Robusto**: Scrittura di codice pronto per essere rilasciato in produzione con una documentazione dell'API interattiva generata automaticamente.
```

* **Basato sugli standard**: Pienamente compatibile con gli open standard per le API: <a href="https://github.com/OAI/OpenAPI-Specification" class="external-link" target="_blank">OpenAPI</a> (prima conosciuto come Swagger) e <a href="http://json-schema.org/" class="external-link" target="_blank">JSON Schema</a>.

> <small>* Questa stima è basata su test eseguiti all'interno di un team di sviluppo di applicazioni professionali.</small>

Reviewer 3: ```suggestion
<small>* Stima basata sull'esito di test eseguiti da un team di sviluppo per applicazioni professionali.</small>
```

</div>

> Avrai anche bisogno di un server ASGI per il rilascio in produzione come <a href="http://www.uvicorn.org" class="external-link" target="_blank">Uvicorn</a> oppure <a href="https://gitlab.com/pgjones/hypercorn" class="external-link" target="_blank">Hypercorn</a>.

Reviewer 3: ```suggestion
Per il rilascio in produzione, sarà necessario un server ASGI come <a href="http://www.uvicorn.org" class="external-link" target="_blank">Uvicorn</a> oppure <a href="https://gitlab.com/pgjones/hypercorn" class="external-link" target="_blank">Hypercorn</a>.
```

    return {"item_id": item_id, "q": q}
```

**Nota**:

> Se vuoi approfondire, consulta la sezione _"In a hurry?"_ su <a href="https://fastapi.tiangolo.com/async/#in-a-hurry" target="_blank">`async` e `await` nella documentazione</a>.

Reviewer: _"Di fretta?"_

</div>

<details markdown="1">
<summary>Informazioni sul comando <code>uvicorn main:app --reload</code>...</summary>

Vediamo il comando `uvicorn main:app` in dettaglio:

* `main`: il `main.py` (il "modulo" Python).

> * `app`: l'oggetto creato dentro `main.py` con la riga di codice `app = FastAPI()`.

Reviewer 3: ```suggestion
* `app`: l'oggetto istanziato all'interno del `main.py` tramite l'istruzione `app = FastAPI()`.
```

* `--reload`: ricarica il server se vengono rilevati cambiamenti del codice. Usalo solo durante la fase di sviluppo.

</details>

### Testa l'API

Apri il browser all'indirizzo <a href="http://127.0.0.1:8000/items/5?q=somequery" class="external-link" target="_blank">http://127.0.0.1:8000/items/5?q=somequery</a>.

Vedrai la seguente risposta JSON:

```JSON
{"item_id": 5, "q": "somequery"}
```

Hai appena creato un'API che:

> * Riceve richieste HTTP ai _paths_ `/` e `/items/{item_id}`.

Reviewer 3: ```suggestion
* Riceve richieste HTTP sui _paths_ `/` e `/items/{item_id}`.
```

> * Entrambi i _paths_ ricevono `GET` <em>operations</em> (anche chiamati <abbr title="metodi HTTP">HTTP _methods_</abbr>).

Reviewer 3: ```suggestion
* Entrambi i _paths_ accettano`GET` <em>operations</em> (conosciuti anche come <abbr title="metodi HTTP">HTTP _methods_</abbr>).
```

* Il _path_ `/items/{item_id}` ha un _path parameter_ `item_id` che deve essere un `int`.
* Il _path_ `/items/{item_id}` ha una `str` _query parameter_ `q`.

### Documentazione interattiva dell'API

Adesso vai su <a href="http://127.0.0.1:8000/docs" class="external-link" target="_blank">http://127.0.0.1:8000/docs</a>.

Vedrai la documentazione interattiva dell'API (offerta da <a href="https://github.com/swagger-api/swagger-ui" class="external-link" target="_blank">Swagger UI</a>):

![Swagger UI](https://fastapi.tiangolo.com/img/index/index-01-swagger-ui-simple.png)

### Documentazione interattiva alternativa

Adesso accedi su <a href="http://127.0.0.1:8000/redoc" class="external-link" target="_blank">http://127.0.0.1:8000/redoc</a>.

Vedrai la documentazione interattiva dell'API (offerta da <a href="https://github.com/Rebilly/ReDoc" class="external-link" target="_blank">ReDoc</a>):

![ReDoc](https://fastapi.tiangolo.com/img/index/index-02-redoc-simple.png)

> ## Esempio più avanzato

Reviewer 2: Maybe "Esempio di aggiornamento" could be better for the context.

Adesso modifica il file `main.py` per ricevere un _body_ da una richiesta `PUT`.

> Dichiara il _body_ usando le annotazioni di tipo standard di Python, grazie a Pydantic.

Reviewer 2: Suggestion: "le annotazioni ~~di tipo~~ standard"

    return {"item_name": item.name, "item_id": item_id}
```

Il server dovrebbe ricaricarsi in automatico (perché hai specificato `--reload` al comando `uvicorn` qui sopra).

### Aggiornamento della documentazione interattiva

Adesso vai su <a href="http://127.0.0.1:8000/docs" class="external-link" target="_blank">http://127.0.0.1:8000/docs</a>.

* La documentazione interattiva dell'API verrà automaticamente aggiornata, includendo il nuovo _body_:

![Swagger UI](https://fastapi.tiangolo.com/img/index/index-03-swagger-02.png)

* Fai click sul pulsante "Try it out", che ti permette di inserire i parametri per interagire direttamente con l'API:

![Swagger UI interaction](https://fastapi.tiangolo.com/img/index/index-04-swagger-03.png)

* Successivamente, fai click sul pulsante "Execute". L'interfaccia utente comunicherà con la tua API, invierà i parametri, riceverà i risultati della richiesta, e li mostrerà sullo schermo:

![Swagger UI interaction](https://fastapi.tiangolo.com/img/index/index-05-swagger-04.png)

> ### Aggiornamento della documentazione alternativa

Reviewer 3: I am not convinced :/

Ora vai su <a href="http://127.0.0.1:8000/redoc" class="external-link" target="_blank">http://127.0.0.1:8000/redoc</a>.

* Anche la documentazione alternativa dell'API mostrerà il nuovo parametro della query e il _body_:

![ReDoc](https://fastapi.tiangolo.com/img/index/index-06-redoc-02.png)

> ### Ricapitolando

Reviewer 3: ```suggestion
### Riepilogo
```

Ricapitolando, è sufficiente dichiarare **una sola volta** i tipi dei parametri, del body, ecc. come parametri di funzioni.

> Puoi farlo usando le annotazioni di tipo standard di Python.

Reviewer 2: Suggestion: "le annotazioni ~~di tipo~~ standard"

> Non c'è bisogno di imparare una nuova sintassi, metodi o classi specifici a una libreria, ecc.

Reviewer 2: "Non c'è bisogno di imparare nuovi metodi, sintassi o classi di una specifica libreria, ecc."
[reference](https://www.impariamoinsieme.com/aggettivo-e-concordanza-con-il-nome/#:~:text=Gli%20aggettivi%20concordano%20sempre%20nel,pi%C3%B9%20il%20discorso%20%C3%A8%20diverso.&text=se%20i%20nomi%20sono%20tutti,concorda%20al%20femminile%20plurale.)

È sufficiente usare **Python 3.6+**.

> Per esempio, per un `int`:

Reviewer 3: ```suggestion
Per esempio, per il tipo `int`:
```

    * Cookies.
    * Headers.
    * Form.
    * File.
* <abbr title="also known as: serialization, parsing, marshalling">Conversione</abbr> dei dati di output: converte dati e tipi di Python a dati per la rete (come JSON):
    * Converte i tipi di Python (`str`, `int`, `float`, `bool`, `list`, ecc).
    * Oggetti `datetime`.
    * Oggetti `UUID`.
    * Modelli del database.
    * ...e molto di più.
* Generazione di una documentazione dell'API interattiva, con scelta dell'interfaccia grafica:
    * Swagger UI.
    * ReDoc.

---

Tornando al precedente esempio, **FastAPI**:

* Validerà che c'è un `item_id` nel percorso delle richieste `GET` e `PUT`.
* Validerà che `item_id` è di tipo `int` per le richieste `GET` e `PUT`.
    * Se non lo è, il client vedrà un errore chiaro e utile.
* Controllerà se c'è un parametro opzionale chiamato `q` (per esempio `http://127.0.0.1:8000/items/foo?q=somequery`) per le richieste `GET`.
    * Siccome il parametro `q` è dichiarato con `= None`, è opzionale.
    * Senza il `None` sarebbe stato obbligatorio specificarlo (come lo è il corpo della richiesta `PUT`).
* Per le richieste `PUT` su `/items/{item_id}`, leggerà il corpo come JSON e

>     * Controlle che abbia un attributo obbligatorio `name`, e che sia una `str`.

Reviewer 2: "Controlla" or "Controllerà"

    * Controlla che abbia un attributo obbligatorio `price`, e che sia un `float`.
    * Controlla che abbia un attributo opzionale `is_offer`, e che sia un `bool`, se presente.
    * Tutto questo funzionerebbe anche con oggetti JSON altamente nidificati.
* Convertirà da JSON a JSON automaticamente.
* Documenterà tutto con OpenAPI, che può essere usato per:
    * Sistemi di documentazione interattivi.
    * Sistemi di generazione di codice dal lato client, per molti linguaggi.
* Fornirà 2 interfacce di documentazione dell'API interattive.

---

> Questa è solo la punta dell'iceberg, ma dovresti avere già un'idea di come il tutto funziona.

Reviewer 3: ```suggestion
Questa è solo la punta dell'iceberg, ma si dovrebbe avere già un'idea di come il tutto funziona.
```

Trova questa riga di codice:

```Python
    return {"item_name": item.name, "item_id": item_id}
```

e cambiala da...

```Python
        ... "item_name": item.name ...
```

...a questa:

```Python
        ... "item_price": item.price ...
```

> ...e osserva come il tuo editor di testo autocompleterà gli attributi e saprà i loro tipi:

Reviewer 3: ```suggestion
...e osserva come il tuo editor di testo autocompleterà gli attributi e sarà in grado di conoscere i loro tipi:
```
